tools/clang/test/taef_exec/lit.cfg

```diff
@@ -0,0 +1,68 @@
+# -*- Python -*-
+
+# Configuration file for the 'lit' test runner.
+
+import os
+import platform
+
+import lit.formats
+import lit.util
+
+# name: The name of this test suite.
+config.name = 'clang-taef-exec'
+
+# Check that the object root is known.
+te = getattr(config, 'te', None)
+if te is None:
+    # Otherwise, we haven't loaded the site specific configuration (the user is
+    # probably trying to run on a test file directly, and either the site
+    # configuration hasn't been created by the build system, or we are in an
+    # out-of-tree build situation).
+
+    # Check for 'clang_taef_exec_site_config' user parameter, and use that if available.
+    site_cfg = lit_config.params.get('clang_taef_exec_site_config', None)
+    if site_cfg and os.path.exists(site_cfg):
+        lit_config.load_config(config, site_cfg)
+        raise SystemExit
+
+# TAEF only runs on Windows
+if platform.system() != 'Windows':
+  config.unsupported = True
+else:
+  test_dll = os.path.join(config.llvm_obj_root, config.llvm_build_mode, 'bin', 'ExecHLSLTests.dll')
+
+  hlsl_data_dir = os.path.join(config.llvm_src_root, 'tools', 'clang', 'unittests', 'HLSLExec')
+
+  test_dir = os.path.join(config.llvm_obj_root, config.llvm_build_mode, 'test')
+  param_hlsl_data_dir = str.format('/p:HlslDataDir={}', hlsl_data_dir)
+
+  extra_params = ["/p:\"ExperimentalShaders=*\"", param_hlsl_data_dir]
+
+  verbose = lit_config.params.get('verbose', None)
+  if verbose == None:
+      extra_params.append('/logOutput:LowWithConsoleBuffering')
+
+  arch = getattr(config, 'taef_arch', None)
+  arch_filter = str.format("@Architecture='{}'", arch)
+  priority_filter = "@Priority<2"
+  future_filter = lit_config.params.get('future_filter', None)
+  if future_filter:
+      priority_filter = "@Priority=2"
```

Comment:
In hcttest.cmd, the `exec-future-filter` combines `exec-future` and `exec-filter` mode.  The first sets the `@Priority` filter, but the second is supposed to set the `@Name` filter.

For this, I think it would be best to have separate options: `taef_exec_future` which just sets this priority, and if we want to support name filtering (that would certainly be important if moving off `hcttest.cmd`), then we should have another `taef_exec_filter` option.  This implies we should have a `taef_filter` option for the other non-exec taef test as well.  Then name filtering needs to be incorporated into the `/select:` string option.

Reply:
For lit test, each name is tested separately with /name:testName.
The testName is get from te /list.
So @Name filter is not offered for lit test.
Maybe we can keep hcttest.com for case @Name filter is needed?

Reply:
I think `future_filter` isn't the best name to go with here, and could either be made consistent with the other taef test as `no_priority` removing the priority filte, or we could name it `exec_future` if we still want to filter to `@Priority=2`.

Another option for the filtering is to provide a `taef_select` argument (and maybe a separate `taef_exec_select`?) that supplies taef's `/select` argument contents instead of constructing them from other arguments here.